src/storage/sql/mod.rs

```diff
@@ -5,33 +5,42 @@ use std::convert::From;
 
 use diesel::pg::upsert;
 use diesel::prelude::*;
-use diesel::r2d2::{ConnectionManager, Pool, PoolError};
 use diesel::result::{DatabaseErrorKind, Error::DatabaseError, Error::NotFound};
+use diesel_async::pooled_connection::deadpool::{BuildError, Pool, PoolError};
+use diesel_async::pooled_connection::AsyncDieselConnectionManager;
+use diesel_async::scoped_futures::ScopedFutureExt;
+use diesel_async::{AsyncConnection, AsyncPgConnection, RunQueryDsl};
 
 use super::{errors::StorageError, Direction, ListSnippetsQuery, Snippet, Storage};
 use schema::{changesets, snippets, tags};
 
+// deadpool's default pool size is # of CPUs * 4, which might be too low. Instead,
+// let's default to PostgreSQL's max_connections value (100) and reserve a small
+// buffer for "admin" connections (like periodic database backups or local sessions).
+const MAX_POOL_SIZE: usize = 96;
```

Comment:
Since PostgreSQL `max_connections` can be changed, do you think we should make this option configurable too?

Reply:
Yeah, let's do that as a follow up!